include/pybind11/detail/internals.h

```diff
@@ -313,12 +313,19 @@ PYBIND11_NOINLINE inline internals &get_internals() {
     return **internals_pp;
 }
 
-/// Works like `internals.registered_types_cpp`, but for module-local registered types:
-inline type_map<type_info *> &registered_local_types_cpp() {
-    static type_map<type_info *> locals{};
-    return locals;
+
+struct local_internals {
+  type_map<type_info *> registered_local_types_cpp;
```

Comment:
Answering the question you asked in a different comment:
I'd prefer the 2nd alternative you pointed out there, with `local` **not** repeated inside the struct.
But I'd approve either.

Reply:
Names have been changed, since everyone seems to prefer them without the `local`